src/engine/client/cl_cgame.cpp

```diff
@@ -1102,6 +1102,32 @@ void CGameVM::QVMSyscall(int syscallNum, Util::Reader& reader, IPC::Channel& cha
 			});
 			break;
 
+		case CG_CM_MARKFRAGMENTSARRAY:
+			IPC::HandleMsg<CMMarkFragmentsArrayMsg>(channel, std::move(reader), [this] (
+				const std::vector<fragmentPacket_t>& fragmentPacketsArray,
+				std::vector<fragmentBuffer_t>& fragmentBuffersArray)
+			{
+				size_t numMarks = fragmentPacketsArray.size();
+
+				fragmentBuffersArray.resize(numMarks);
+
+				for ( size_t m = 0; m < numMarks; m++ )
+				{
+					const fragmentPacket_t *fragmentPacket = &fragmentPacketsArray[ m ];
+					fragmentBuffer_t *fragmentBuffer = &fragmentBuffersArray[ m ];
+
+					fragmentBuffer->numFragments = re.MarkFragments(
+						4,
```

Comment:
Note: The rewriting of `re.MarkFragments` to not uselessly expects global constants as input like `4`, `MAX_MARK_POINTS`, and `MAX_MARK_FRAGMENTS` will be done in a future PR after NUKING the legacy `trap_CM_MarkFragments` function where the constants were written in game and sent to engine instead.